File: 02.AbstractFactory/java/src/abstractFactory/ShapeFactory.java
package abstractFactory;

import shapes.*;
import colors.*;

public class ShapeFactory extends AbstractFactory {

	public Shape getShape(String shapeType){
		
		if ( shapeType == null){
			return null;			
		}
		
		if (shapeType.equalsIgnoreCase("CIRCLE")) {
			return new Circle();			
		}else if (shapeType.equalsIgnoreCase("SQUARE")) {
			return new Square();
		}else if(shapeType.equalsIgnoreCase("RECTANGLE")){
			return new Rectangle();			
		}
		
		return null;
	}
	
	public Color getColor(String color) {
		return null;
	}
	
}

File: 02.AbstractFactory/cpp/src/factory/shapeFactory.cpp
#pragma once
#include "shapeFactory.h"
#include <iostream>
#include <string>
#include "../shape/shape.h" 
#include "../shape/circle.cpp" 
#include "../shape/square.cpp" 
#include "../shape/rectangle.cpp" 

shapeFactory::shapeFactory() {
}

shapeFactory::shapeFactory(const shapeFactory& orig) {
}

shapeFactory::~shapeFactory() {
}

shape* shapeFactory::getShape(std::string shapeChosen){
    
    std::cout << "Getting the shape from the factory." << std::endl;
    if (shapeChosen == "circle") {
        return new circle();
    }else if (shapeChosen == "square"){
        return new square();
    }else if (shapeChosen == "rectangle"){
        return new rectangle();
    }
    
}
